src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1441,11 +1441,41 @@ def vnc_bind_choices(self):
                     'datastore.update', self._config.datastore, device['id'], device
                 )
 
+    @private
+    async def disk_uniqueness_integrity_check(self, device, vm):
+        # This ensures that the disk is not already present for `vm`
+        def translate_device(dev):
+            # A disk should have a path configured at all times, when that is not the case, that means `dtype` is DISK
+            # and end user wants to create a new zvol in this case.
+            return dev['attributes'].get('path') or f'/dev/zvol/{dev["attributes"]["zvol_name"]}'
+
+        disks = [
+            d for d in vm['devices']
+            if d['dtype'] in ('DISK', 'RAW', 'CDROM') and translate_device(d) == translate_device(device)
+        ]
+        if not disks:
+            # We don't have that disk path in vm devices, we are good to go
+            return True
+        elif len(disks) > 1:
```

Comment:
Why a VM cannot have more than 1 disk?

Reply:
A VM cannot have more then 1 disk having the same path

Reply:
Can you show me where its being checked for same path?

Reply:
https://github.com/freenas/freenas/pull/3560/files#diff-2624b1c1bd7ddfbcb5b2ddf3655d961eR1454